src/compiler/parser.ts

```diff
@@ -436,6 +436,13 @@ module ts {
         // attached to the EOF token.
         let parseErrorBeforeNextFinishedNode: boolean = false;
 
+        export const enum StatementFlags {

```

Comment:
I tried and it just doesn't make sense to use ParsingContext. We can discuss in person if you'd like.